Fix Type Annotations in pandas/_version.py

Part of #25882 

Current errors are as follows:

    pandas\_version.py:51: error: Need type annotation for 'LONG_VERSION_PY'
    pandas\_version.py:52: error: Need type annotation for 'HANDLERS'

--- a/mypy.ini
+++ b/mypy.ini
@@ -5,9 +5,6 @@ follow_imports=silent
 [mypy-pandas.conftest,pandas.tests.*]
 ignore_errors=True
 
-[mypy-pandas._version]
-ignore_errors=True
-
 [mypy-pandas.compat]
 ignore_errors=True
 

--- a/pandas/_version.py
+++ b/pandas/_version.py
@@ -12,6 +12,7 @@ import os
 import re
 import subprocess
 import sys
+from typing import Dict
 
 from pandas.compat import PY3
 
@@ -48,8 +49,8 @@ class NotThisMethod(Exception):
     pass
 
 
-LONG_VERSION_PY = {}
-HANDLERS = {}
+LONG_VERSION_PY = {}  # type: Dict
+HANDLERS = {}  # type: Dict[str, Dict]
 
 
 def register_vcs_handler(vcs, method):  # decorator

--- a/pandas/_version.py
+++ b/pandas/_version.py
@@ -12,7 +12,7 @@ import os
 import re
 import subprocess
 import sys
-from typing import Dict
+from typing import Callable, Dict
 
 from pandas.compat import PY3
 
@@ -49,11 +49,11 @@ class NotThisMethod(Exception):
     pass
 
 
-LONG_VERSION_PY = {}  # type: Dict
-HANDLERS = {}  # type: Dict[str, Dict]
+# LONG_VERSION_PY = {}  # type: Dict
+HANDLERS = {}  # type: Dict[str, Dict[str, Callable]]
 
 
-def register_vcs_handler(vcs, method):  # decorator
+def register_vcs_handler(vcs: str, method: str) -> Callable:  # decorator
     def decorate(f):
         if vcs not in HANDLERS:
             HANDLERS[vcs] = {}

--- a/pandas/_version.py
+++ b/pandas/_version.py
@@ -49,12 +49,11 @@ class NotThisMethod(Exception):
     pass
 
 
-# LONG_VERSION_PY = {}  # type: Dict
 HANDLERS = {}  # type: Dict[str, Dict[str, Callable]]
 
 
 def register_vcs_handler(vcs: str, method: str) -> Callable:  # decorator
-    def decorate(f):
+    def decorate(f: Callable) -> Callable:
         if vcs not in HANDLERS:
             HANDLERS[vcs] = {}
         HANDLERS[vcs][method] = f

